ships/0025-strategy-parameters-enhancements.md

```diff
@@ -0,0 +1,260 @@
+<!--
+Copyright The Shipwright Contributors
+
+SPDX-License-Identifier: Apache-2.0
+-->
+
+---
+title: buildstrategy-parameter-enhancements
+authors:
+  - @SaschaSchwarze0
+reviewers:
+  - @HeavyWombat
+  - @GabeMontero
+  - @ImJasonH
+approvers:
+  - @adambkaplan
+creation-date: 2021-10-06
+last-updated: 2021-10-06
+status: implementable
+---
+
+# Surface Step Results in BuildRun Status
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [docs](/docs/)
+
+## Open Questions [optional]
+
+None
+
+## Summary
+
+Recently, we introduced build strategy parameters with default values support through [ship 0014](0014-parameterize-strategies.md). This only supported single strings as values.
+
+This ship proposes an extension to support arrays and secret values for strategy parameters.
```

Comment:
do we want to allow confiig map value from's as well .... it would bring us to conceptual equality to what we just dropped for env var support

Reply:
Yeah, just thinking about it. One famous use case for `ARGs` are versions of utilities that one downloads through `wget` (like `ko`). Sharing that version across builds and therefore having it in a ConfigMap makes sense. And implementation-wise, the same approach as for secrets would work, I think. I'll update it.

Reply:
@gabemontero I added `configMapValue` as additional option.

What's your thinking on the implementation for ConfigMap key references. Should we ...

1. ... also use an environment variable to pass the value
2. ... instead directly read the ConfigMap value and store it in the Tekton param value

The difference is imo clear: secret values need protection. We therefore cannot take a value from there and use it directly in the TaskRun. For a ConfigMap, this should be possible. We could therefore use option (2). On the other hand, (1) would be functionally equivalent and use the same or very similar code base for both kinds.